Android/PebbleKitSportsAPIDemo/app/src/main/java/com/pebble/pebblekitsportsapidemo/MainActivity.java

```diff
@@ -52,29 +59,45 @@ public void onClick(View v) {
         });
 
         // Add Configure Sports button listeners
-        final Button sportsDummyData = (Button)findViewById(R.id.dummy_data_sports);
+        final Button sportsDummyData = (Button) findViewById(R.id.dummy_data_sports);
         sportsDummyData.setOnClickListener(new View.OnClickListener() {
 
             @Override
             public void onClick(View v) {
-                // Send some dummy data
-                PebbleDictionary out = new PebbleDictionary();
-                out.addString(Constants.SPORTS_TIME_KEY, "12:52");
-                out.addString(Constants.SPORTS_DISTANCE_KEY, "23.8");
-                PebbleKit.sendDataToPebble(getApplicationContext(), Constants.SPORTS_UUID, out);
+                final long startTime = System.currentTimeMillis();
+                handler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        double time = (double)(System.currentTimeMillis() - startTime) / 1000;
+                        float accel = (float)Math.abs(-0.01 * Math.sin(time / 10) * 102);
+
+                        currentState.setTimeInSec((int)time);
+                        currentState.setDistance((float)(1.5 * time + Math.sin(time / 10)) / 1000);
+                        currentState.setHeartBPM((byte)(90 + Math.sin(time / 10) * 30));
+                        if (displayPace) {
+                            currentState.setPaceInSec((int)((10 / (15 + Math.cos(time / 10))) * 1000));
+                        } else {
+                            currentState.setSpeed((float)(15 + Math.cos(time/10))/10 * (3600/1000));
+                        }
+                        currentState.setCustomLabel("Accel (in mG)");
```

Comment:
L82 can be moved under L67 or under L37. Doesn’t need to be repeated each iteration of the runnable.